practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/repository/HapiFhirRepository.java

```diff
@@ -21,7 +20,7 @@
     private static final String PRACTITIONER_ENTITY_NAME = "Practitioner";
 
     private final FhirContextWrapper fhirContextWrapper;
-    private final Logger logger = LoggerFactory.getLogger(HapiFhirRepository.class);
+    private Logger logger = LoggerFactory.getLogger(HapiFhirRepository.class);
```

Comment:
Archie, shouldn't this remain

private static final Logger LOGGER

Reply:
Not quite sure about the "static" thing really - I think I saw weird threading issues once with a logger, due to a static instance, so I've avoided static loggers ever since (at this point, that might now be me being superstitious). However, the slf4j FAQ mentions they no longer recommend one approach over the other (http://slf4j.org/faq.html#declared_static). The only one of the advantages that I found interesting was that non-static is more IOC-friendly. Which brings me on to the next bit....

I'm a huge fan of "final", but I dropped it in this case to allow me to add a setLogger method. This seemed to me to be the easiest way to allow dependency injection for the purposes of a test - I use it to set a silent mock logger, so that the test ouput isn't polluted with a distracting stacktrace. I wonder if there's a better way though...

Reply:
I've now removed setLogger and changed this back to final. Sonar wasn't happy about the log-and-throw antipattern it was seeing so I removed the log and put its message into the RestException that is thrown. This is then logged by the global RestException handler.